### DISCUSSION TITLE
Long `status_msg` will corrupt the terminal

### DISCUSSION THREAD

**Discussion by plxty** [helix-editor/helix#11272] [Troubleshooting]
### Summary

For a simple scenario, select a line or a block region that is longer than the terminal's width, use '*' key to save them into the search register, then the terminal is *corrupted*, with many shadow words.

A really simple solution is to limit the `status_msg` length, 

https://github.com/hyptrap/helix/blob/09680f56256c2191e618f26360cf14e63d99db28/helix-term/src/ui/editor.rs#L1515-L1532

But not sure what the root cause really is, IMO after `status_msg` has cleared, the area should be redrawed, and thus may not affects the whole screen.

### Reproduction Steps

<https://asciinema.org/a/7CPdiJgDGa1XFAHAVdhfgP5Ib>

### Helix log

Nope.

### Platform

Linux

### Terminal Emulator

konsole

### Installation Method

zypper

### Helix Version

helix 24.7 (079f5442)

**Comment by woojiq** [comment]
Can't reproduce on helix 24.3 (2cadec0b)
Can't reproduce on helix 24.7
terminal: wezterm

**Comment by pascalkuthe** [comment]
can't reproduce on master either, could be some issue with your terminal emulator (or maybe it's a development build with a stray print?)

**Comment by plxty** [comment]
That's really weird, I can reproduce both in openSUSE KDE and in Windows WSL2 (openSUSE as well), suspecting the distro's issue. Maybe investigating how the openSUSE patched the terminal some time myself, sorry for wasting the time🙁.
